TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +55,76 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
             SyntaxProperty<SelectionModifier> selectionModifier = null;
             if (context.SQL_ALL() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All, ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
             }
             else if (context.SQL_DISTINCT() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct, ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
+
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var dotStarSelections = context.selections().selection().Select(CreateDotStarSelection).Where(selection => selection != null).ToList();
+
+                return new SelectClause(selectionModifier, dotStarSelections);
+            }
+            return null;
+        }
+
+        private TableOrViewAllColumnsSelection CreateDotStarSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.dotStarSelection() != null)
+            {
+                var tableOrViewOrCorrelationName = context.dotStarSelection().tableOrViewOrCorrelationName();
+                Token name = tableOrViewOrCorrelationName.Name as Token;
+                Token schemaName = tableOrViewOrCorrelationName.SchemaName as Token;
+                Token dbms = tableOrViewOrCorrelationName.DBMS as Token;
+                SymbolReference fullName = CreateSymbolReference(name, schemaName, dbms);
```

Comment:
I think here it is better to create a class TableViewCorrelationName that ihnerits from QualifiedSymbolReference.
So that we can add getters on properties like DBMS, SchemaName, or Name.

Reply:
So I keep on thinking let's TableViewCorrelationName inherits from SqlObject rather than QualifiedSymbolReference.
I Suggest to have a class SqlStorageArea that ihnerits from StorageArea in this case TableViewCorrelationName can have an attribute SqlStorageArea built from QualifiedSymbolReference.
![image](https://user-images.githubusercontent.com/19149269/160605057-acda1777-7deb-4fc8-8842-38bcfb6045de.png)